### ISSUE TITLE
Upgrading to Craft 4 - One of Blitz’s migrations failed.

### ISSUE THREAD

**Issue by bymayo** [putyourlightson/craft-blitz#402]
When I upgrade to Craft 4 and run `./craft migrate/all` the migrations stop and error on `One of Blitz’s migrations failed.`

Below is the error it produces.

```
TypeError: craft\helpers\ProjectConfig::unpackAssociativeArray(): Argument #1 ($array) must be of type array, null given, called in /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/putyourlightson/craft-blitz/src/migrations/m220330_120000_update_settings.php on line 39

Migration: putyourlightson\blitz\migrations\m220330_120000_update_settings

Output:

Exception: craft\helpers\ProjectConfig::unpackAssociativeArray(): Argument #1 ($array) must be of type array, null given, called in /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/putyourlightson/craft-blitz/src/migrations/m220330_120000_update_settings.php on line 39 (/Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/helpers/ProjectConfig.php:389)
#0 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/putyourlightson/craft-blitz/src/migrations/m220330_120000_update_settings.php(39): craft\helpers\ProjectConfig::unpackAssociativeArray(NULL)
#1 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/db/Migration.php(49): putyourlightson\blitz\migrations\m220330_120000_update_settings->safeUp()
#2 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/db/MigrationManager.php(233): craft\db\Migration->up(true)
#3 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/db/MigrationManager.php(149): craft\db\MigrationManager->migrateUp(Object(putyourlightson\blitz\migrations\m220330_120000_update_settings))
#4 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/services/Updates.php(252): craft\db\MigrationManager->up()
#5 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(493): craft\services\Updates->runMigrations(Array)
#6 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/controllers/UpdaterController.php(203): craft\controllers\BaseUpdaterController->runMigrations(Array, 'restore-db')
#7 [internal function]: craft\controllers\UpdaterController->actionMigrate()
#8 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#10 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('migrate', Array)
#11 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('updater/migrate', Array)
#12 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/web/Application.php(712): craft\web\Application->runAction('updater/migrate')
#13 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/web/Application.php(258): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#14 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#15 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/public/index.php(26): yii\base\Application->run()
#16 /Users/jasonmayo/.composer/vendor/laravel/valet/server.php(234): require('/Users/jasonmay...')
#17 {main}
```

I've tried removing the `/config/blitz.php` file to see if this had a config setting that is no longer used but still hit the same issue.

### Versions
- Plugin version: 4.0.1
- Craft version: 4.0.0

**Comment by bencroker** [comment]
Thanks for reporting this, fixed in https://github.com/putyourlightson/craft-blitz/commit/da2edf5618f67e8a9061bf50c48a6be72cedc7eb and released in version 4.0.2.

**Comment by bymayo** [comment]
Thanks for fixing that @bencroker but it's hitting another error now when I updated to 4.0.2

```
*** applying m220330_120000_update_settings
Exception: putyourlightson\blitz\migrations\m220330_120000_update_settings::_updateQueryStringParams(): Argument #1 ($queryStringParams) must be of type array, null given, called in /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/putyourlightson/craft-blitz/src/migrations/m220330_120000_update_settings.php on line 49 (/Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/putyourlightson/craft-blitz/src/migrations/m220330_120000_update_settings.php:85)
#0 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/putyourlightson/craft-blitz/src/migrations/m220330_120000_update_settings.php(49): putyourlightson\blitz\migrations\m220330_120000_update_settings->_updateQueryStringParams(NULL)
#1 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/db/Migration.php(49): putyourlightson\blitz\migrations\m220330_120000_update_settings->safeUp()
#2 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/console/controllers/BaseMigrateController.php(758): craft\db\Migration->up()
#3 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/console/controllers/MigrateController.php(360): yii\console\controllers\BaseMigrateController->migrateUp('m220330_120000_...')
#4 [internal function]: craft\console\controllers\MigrateController->actionAll()
#5 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#6 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#7 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/console/Controller.php(182): yii\base\Controller->runAction('all', Array)
#8 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction('all', Array)
#9 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#10 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/craftcms/cms/src/console/Application.php(90): yii\console\Application->runAction('migrate/all', Array)
#11 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#12 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/vendor/yiisoft/yii2/base/Application.php(384): yii\console\Application->handleRequest(Object(craft\console\Request))
#13 /Users/jasonmayo/Code/Github/@madebyshape/craft-cms/craft(27): yii\base\Application->run()
#14 {main}
*** failed to apply m220330_120000_update_settings (time: 0.001s)


22 from 30 migrations were applied.

Migration failed. The rest of the migrations are canceled.
```

**Comment by bencroker** [comment]
Sorry about that, just fixed in version 4.0.3, let me know if that works now.

**Comment by bymayo** [comment]
Works! 🎉

Thanks for fixing that so quickly
